clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchMetricsManager.java

```diff
@@ -101,32 +101,38 @@ void recordRecordsFetched(int records) {
 
     void recordBytesFetched(String topic, int bytes) {
         String name = topicBytesFetchedMetricName(topic);
-        Sensor bytesFetched = new SensorBuilder(metrics, name, () -> topicTags(topic))
-                .withAvg(metricsRegistry.topicFetchSizeAvg)
-                .withMax(metricsRegistry.topicFetchSizeMax)
-                .withMeter(metricsRegistry.topicBytesConsumedRate, metricsRegistry.topicBytesConsumedTotal)
-                .build();
+        maybeRecordDeprecatedBytesFetched(name, topic, bytes);
+
+        Sensor bytesFetched = new SensorBuilder(metrics, name, () -> Map.of("topic", topic))
+            .withAvg(metricsRegistry.topicFetchSizeAvg)
+            .withMax(metricsRegistry.topicFetchSizeMax)
+            .withMeter(metricsRegistry.topicBytesConsumedRate, metricsRegistry.topicBytesConsumedTotal)
+            .build();
         bytesFetched.record(bytes);
     }
 
     void recordRecordsFetched(String topic, int records) {
         String name = topicRecordsFetchedMetricName(topic);
-        Sensor recordsFetched = new SensorBuilder(metrics, name, () -> topicTags(topic))
-                .withAvg(metricsRegistry.topicRecordsPerRequestAvg)
-                .withMeter(metricsRegistry.topicRecordsConsumedRate, metricsRegistry.topicRecordsConsumedTotal)
-                .build();
+        maybeRecordDeprecatedRecordsFetched(name, topic, records);
+
+        Sensor recordsFetched = new SensorBuilder(metrics, name, () -> Map.of("topic", topic))
+            .withAvg(metricsRegistry.topicRecordsPerRequestAvg)
+            .withMeter(metricsRegistry.topicRecordsConsumedRate, metricsRegistry.topicRecordsConsumedTotal)
+            .build();
         recordsFetched.record(records);
     }
 
     void recordPartitionLag(TopicPartition tp, long lag) {
         this.recordsLag.record(lag);
 
         String name = partitionRecordsLagMetricName(tp);
-        Sensor recordsLag = new SensorBuilder(metrics, name, () -> topicPartitionTags(tp))
-                .withValue(metricsRegistry.partitionRecordsLag)
-                .withMax(metricsRegistry.partitionRecordsLagMax)
-                .withAvg(metricsRegistry.partitionRecordsLagAvg)
-                .build();
+        maybeRecordDeprecatedPartitionLag(name, tp, lag);
+
+        Sensor recordsLag = new SensorBuilder(metrics, name, () -> Map.of("topic", tp.topic(), "partition", String.valueOf(tp.partition())))
```

Comment:
nit: Could you please consider placing "topic" before "partition"? See the following screenshot:
![Screenshot From 2024-12-18 17-46-17](https://github.com/user-attachments/assets/85c1d1de-0a18-4d9a-8c46-2d4e0d4be760)

That can be addressed by using `Utils.mkMap`

Reply:
Make sense, I earlier considered it but didn't do as previous tags were not ordered. I have added the change.